src/main/java/noppes/npcs/DataStats.java

```diff
@@ -110,8 +111,9 @@ public NBTTagCompound writeToNBT(NBTTagCompound compound)
 		compound.setBoolean("CanDrown", canDrown);
 		compound.setBoolean("BurnInSun", burnInSun);
 		compound.setBoolean("NoFallDamage", noFallDamage);
-		compound.setBoolean("CanDespawn", canDespawn);
 		compound.setBoolean("AttackInvisible", attackInvisible);
+		compound.setBoolean("CanDespawn", canDespawn);
+		compound.setBoolean("PlayerSetCanDespawn", playerSetCanDespawn);
```

Comment:
What does ``playerSetCanDespawn`` do? I can't find any time where it doesn't equal ``canDespawn``.

Reply:
in `RoleFollower.setOwner()` `npc.stats.canDespawn = false;` to prevent an npc that can naturally despwn from despawning while following a player. in `RoleFollower.killed()` `npc.stats.canDespawn = npc.stats.playerSetCanDespawn;` to set the despawn boolean back to what was set by the player. This is supposed to only do something different if a player sets canDespawn to true in the gui (`SubGuiNpcRespawn.actionPerformed`) or api.

Reply:
Ahh I see what you mean. This is good then yup! I'll mark all these as resolved and the files as viewed 👍